src/cromshell/logs/command.py

```diff
@@ -57,222 +76,281 @@ def main(
-    )
+    for workflow_id in workflow_ids:
+        command_setup_utils.resolve_workflow_id_and_server(
+            workflow_id=workflow_id, cromshell_config=config
+        )
 
-    return 0
+        task_logs = get_task_level_outputs(
+            config,
+            requested_status=status_param,
+            expand_subworkflows=not dont_expand_subworkflows,
+        )
 
+        if fetch_logs:
+            download_task_level_logs(all_task_log_metadata=task_logs)
+        else:
+            if json_summary:
+                io_utils.pretty_print_json(format_json=task_logs)
+            else:
+                print_task_level_logs(
+                    all_task_log_metadata=task_logs, cat_logs=print_logs
+                )
 
-def check_workflow_for_calls(workflow_status_json: dict) -> None:
-    """Check if the workflow has any calls"""
+    return return_code
 
-    if not workflow_status_json.get("calls"):
-        if workflow_status_json.get("failures"):
-            LOGGER.error(
-                "Empty 'calls' key found in workflow metadata. "
-                "Workflow failed with the following error(s): %s"
-                % workflow_status_json["failures"],
-            )
-            raise KeyError(
-                "Empty 'calls' key found in workflow metadata. "
-                "Workflow failed with the following error(s): %s"
-                % workflow_status_json["failures"],
-            )
-        else:
-            LOGGER.error(
-                "Empty 'calls' key found in workflow metadata. "
-                "This may indicate no tasks were run by the workflow, "
-                "the workflow has yet to run any tasks, or "
-                "a failure occurred before the workflow started."
-            )
-            raise KeyError(
-                "Empty 'calls' key found in workflow metadata."
-                "This may indicate no tasks were run by the workflow, "
-                "the workflow has yet to run any tasks, or "
-                "a failure occurred before the workflow started."
-            )
 
+def get_task_level_outputs(config, expand_subworkflows, requested_status) -> dict:
+    """Get the task level outputs from the workflow metadata
 
-def obtain_and_print_logs(
-    config,
-    metadata_param: list,
-    print_logs: bool,
-    status_params: list,
-    dont_expand_subworkflows: bool,
-) -> None:
-    """Format metadata parameters and obtains metadata from cromwell server"""
+    Args:
+        config (object): The cromshell config object
+        requested_status (list): The list of requested status
+        expand_subworkflows (bool) : Whether to expand subworkflows
+    """
 
-    # Combine keys and flags into a dictionary
+    metadata_keys = [
+        "id",
+        "executionStatus",
+        "subWorkflowMetadata",
+        "subWorkflowId",
+        "failures",
+        "attempt",
+        "backendLogs",
+        "backend",
+        "shardIndex",
+        "stderr",
```

Comment:
"As of now Azure does not provide backendlogs, hence retrieving stderr and stdout". add as a comment

Reply:
Add the above comment as inline comment